std/range/primitives.d

```diff
@@ -161,17 +161,12 @@ Params:
 Returns:
     true if R is an InputRange, false if not
  */
-template isInputRange(R)
-{
-    enum bool isInputRange = is(typeof(
-    (inout int = 0)
-    {
-        R r = R.init;     // can define a range object
-        if (r.empty) {}   // can test for empty
-        r.popFront;       // can invoke popFront()
-        auto h = r.front; // can get the front of the range
-    }));
-}
+enum bool isInputRange(R) =
+    is(typeof(R.init) == R)
+    && is(ReturnType!((R r) => r.empty) == bool)
+    && is(typeof((R r) => r.front))
+    && !is(ReturnType!((R r) => r.front) == void)
+    && is(typeof((R r) => r.popFront));
```

Comment:
how about already copying over the plain-text descriptions as comments, s.t. they aren't lost & it's easy to switch to whatever pretty-print mechanism works best, e.g.

```d
      is(typeof(R.init) == R)  // can define a range object
&& is(ReturnType!((R r) => r.empty) == bool) // can test for empty
```